src/lib/components/EventStore/index.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../EventScheduleModal';
+
+import {store} from '../../store';
+
+// Pretend the date is active within this buffer. This is useful so we can
+// reveal the player for a bit more time.
+const bufferHours = 1;
+
+/**
+ * @fileoverview Publishes event data to Unistore.
+ */
+
+class EventStore extends HTMLElement {
+  constructor() {
+    super();
+    this.onStateChanged = this.onStateChanged.bind(this);
+
+    this._timeOffset = 0;
+    this._activeDay = null;
+    this._data = [];
+  }
+
+  /**
+   * @param {boolean=} change this should force a change
+   */
+  _update(change = false) {
+    const now = +new Date() + this._timeOffset;
+
+    // If there was a previously active day (because the user has their tab open for a long time),
+    // then prefer it over showing the upcoming day (we store that in nextPendingDay).
+    let nextPendingDay = null;
+
+    for (const day of this._data) {
+      const timeOffsetBy = (hours) => {
+        const d = new Date(day.when);
+        d.setHours(d.getHours() + hours);
+        return +d;
+      };
+
+      const bufferStart = timeOffsetBy(-bufferHours);
+      const bufferEnd = timeOffsetBy(day.duration + bufferHours);
```

Comment:
Maybe another way of doing this would be to make bufferEnd = (utcTimeOfNextDay - 1hr). If we do that then you might be able to remove the nextPending day stuff?

Reply:
No, because we have several ranges.

a. Within day 1
b. Ambiguous time between days
c. Within day 2
d. Ambiguous time between days
e. Within day 3

So the end is around each individual day.